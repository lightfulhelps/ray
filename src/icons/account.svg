<svg width="24" height="24" viewBox="0 0 24 24" fill="#ADB5BD" xmlns="http://www.w3.org/2000/svg">
<path d="M12 2C6.47715 2 2 6.47715 2 12C2 17.5228 6.47715 22 12 22C17.5228 22 22 17.5228 22 12C22 9.34784 20.9464 6.8043 19.0711 4.92893C17.1957 3.05357 14.6522 2 12 2ZM12 5C13.6569 5 15 6.34315 15 8C15 9.65686 13.6569 11 12 11C10.3431 11 9 9.65686 9 8C9 6.34315 10.3431 5 12 5ZM17.62 16.16C16.2994 17.9384 14.2151 18.9868 12 18.9868C9.78493 18.9868 7.70056 17.9384 6.38 16.16C6.17921 15.8633 6.15251 15.4818 6.31 15.16L6.52 14.72C7.01488 13.671 8.07012 13.0012 9.23 13H14.77C15.9136 13.0015 16.9568 13.6531 17.46 14.68L17.69 15.15C17.8514 15.4745 17.8246 15.8609 17.62 16.16Z" fill="#1E2128"/>
</svg>
