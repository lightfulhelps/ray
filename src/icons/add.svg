<?xml version="1.0" encoding="UTF-8"?>
<svg width="24px" height="24px" viewBox="0 0 24 24" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <!-- Generator: Sketch 51.2 (57519) - http://www.bohemiancoding.com/sketch -->
    <title>add</title>
    <desc>Created with Sketch.</desc>
    <defs></defs>
    <g id="add" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
        <path d="M13.1,10.9 L19,10.9 C19.5522847,10.9 20,11.3477153 20,11.9 L20,12.1 C20,12.6522847 19.5522847,13.1 19,13.1 L13.1,13.1 L13.1,19 C13.1,19.5522847 12.6522847,20 12.1,20 L11.9,20 C11.3477153,20 10.9,19.5522847 10.9,19 L10.9,13.1 L5,13.1 C4.44771525,13.1 4,12.6522847 4,12.1 L4,11.9 C4,11.3477153 4.44771525,10.9 5,10.9 L10.9,10.9 L10.9,5 C10.9,4.44771525 11.3477153,4 11.9,4 L12.1,4 C12.6522847,4 13.1,4.44771525 13.1,5 L13.1,10.9 Z" id="shape" fill="#ADB5BD" fill-rule="nonzero"></path>
    </g>
</svg>