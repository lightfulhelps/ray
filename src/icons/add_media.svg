<?xml version="1.0" encoding="UTF-8"?>
<svg width="24px" height="24px" viewBox="0 0 24 24" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <!-- Generator: Sketch 51.2 (57519) - http://www.bohemiancoding.com/sketch -->
    <title>add_media</title>
    <desc>Created with Sketch.</desc>
    <defs></defs>
    <g id="add_media" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
        <path d="M4.10227083,3.52 L3.49822917,2.686375 C4.1173125,2.2375 4.84960417,2 5.6159375,2 L6.0505625,2 L6.0505625,3.02995833 L5.6159375,3.02995833 C5.06810417,3.02995833 4.5448125,3.199375 4.10227083,3.52 Z M3.0295625,7.43479167 L2.00039583,7.43479167 L2.00039583,5.61554167 C2.00039583,5.401 2.01939583,5.18408333 2.05739583,4.97270833 L3.07072917,5.15320833 C3.0438125,5.303625 3.0295625,5.458 3.0295625,5.61554167 L3.0295625,7.43479167 Z M2.33052083,12.35975 C2.11122917,11.8839583 2.00039583,11.3772917 2.00039583,10.8508333 L2.00039583,9.80979167 L3.0295625,9.80979167 L3.0295625,10.8508333 C3.0295625,11.226875 3.10872917,11.5894583 3.2646875,11.9282917 L2.33052083,12.35975 Z M13.8928125,12.8015 L13.0275208,12.2449583 C13.2958958,11.8293333 13.4368125,11.3472083 13.4368125,10.8508333 L13.4368125,10.2444167 L14.4659792,10.2444167 L14.4659792,10.8508333 C14.4659792,11.545125 14.2680625,12.219625 13.8928125,12.8015 Z M6.82877083,14.466375 L5.6159375,14.466375 C5.1599375,14.466375 4.71502083,14.3816667 4.29227083,14.2154167 L4.66989583,13.2582917 C4.97152083,13.37625 5.28977083,13.4364167 5.6159375,13.4364167 L6.82877083,13.4364167 L6.82877083,14.466375 Z M10.8599375,14.466375 L10.8488542,14.466375 L9.20377083,14.466375 L9.20377083,13.4364167 L10.8504375,13.4364167 C11.0863542,13.4577917 11.2557708,13.41425 11.4513125,13.36675 L11.6919792,14.3674167 C11.4196458,14.433125 11.1401875,14.466375 10.8599375,14.466375 Z" id="Fill-1" fill="#ADB5BD" transform="translate(8.233188, 8.233187) rotate(-270.000000) translate(-8.233188, -8.233187) "></path>
        <path d="M16.2056667,5.1670625 C18.6075833,5.1670625 20.5614167,7.12089583 20.5614167,9.5228125 L20.5614167,16.2060625 C20.5614167,18.6087708 18.6075833,20.5618125 16.2056667,20.5618125 L9.52241667,20.5618125 C7.1205,20.5618125 5.16666667,18.6087708 5.16666667,16.2060625 L5.16666667,9.5228125 C5.16666667,7.12089583 7.1205,5.1670625 9.52241667,5.1670625 L16.2056667,5.1670625 Z M9.5213875,6.40522917 C7.80347083,6.40522917 6.40459583,7.80410417 6.40459583,9.5228125 L6.40459583,13.5579375 L7.1511375,12.8121875 C8.2246375,11.7394792 9.96392917,11.7394792 11.0374292,12.8121875 L11.7063875,13.4803542 L14.7891375,10.3976042 C15.8626375,9.32410417 17.6027208,9.32410417 18.6770125,10.3976042 L19.3230125,11.0436042 L19.3230125,9.5228125 C19.3230125,7.80410417 17.9241375,6.40522917 16.2054292,6.40522917 L9.5213875,6.40522917 Z" id="Combined-Shape" fill="#ADB5BD"></path>
    </g>
</svg>