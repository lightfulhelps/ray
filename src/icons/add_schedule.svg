<?xml version="1.0" encoding="UTF-8"?>
<svg width="24px" height="24px" viewBox="0 0 24 24" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <!-- Generator: Sketch 51.2 (57519) - http://www.bohemiancoding.com/sketch -->
    <title>add_schedule</title>
    <desc>Created with Sketch.</desc>
    <defs></defs>
    <g id="add_schedule" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
        <path d="M17.2213451,5.38621639 L16.3144932,5.38621639 L16.3144932,4.7506587 C16.3144932,4.33608074 15.9752078,4 15.5566782,4 C15.1381486,4 14.7988632,4.33608074 14.7988632,4.7506587 L14.7988632,5.38621639 L9.20113671,5.38621639 L9.20113671,4.7506587 C9.20113671,4.33608074 8.86185139,4 8.44332175,4 C8.0247921,4 7.68550678,4.33608074 7.68550678,4.7506587 L7.68550678,5.38621639 L6.77865487,5.38621639 C5.24404616,5.38621639 4,6.6185147 4,8.13863163 L4,17.3567223 C4,18.8768392 5.24404616,20.1091375 6.77865487,20.1091375 L17.2213451,20.1091375 C18.7559538,20.1091375 20,18.8768392 20,17.3567223 L20,8.13863163 C20,6.6185147 18.7559538,5.38621639 17.2213451,5.38621639 Z M6.78118092,6.88753379 L7.68803283,6.88753379 L7.68803283,8.02353062 C7.68803283,8.43810858 8.02731815,8.77418932 8.4458478,8.77418932 C8.86437744,8.77418932 9.20366276,8.43810858 9.20366276,8.02353062 L9.20366276,6.88753379 L14.7988632,6.88753379 L14.7988632,8.02353062 C14.7988632,8.43810858 15.1381486,8.77418932 15.5566782,8.77418932 C15.9752078,8.77418932 16.3144932,8.43810858 16.3144932,8.02353062 L16.3144932,6.88753379 L17.2213451,6.88753379 C17.9188945,6.88753379 18.4843701,7.44766978 18.4843701,8.13863163 L18.4843701,10.4156303 L5.51562993,10.4156303 L5.51562993,8.13863163 C5.51562993,7.80681985 5.64869825,7.48859844 5.88556137,7.25397244 C6.1224245,7.01934559 6.44367983,6.88753379 6.77865487,6.88753379 L6.78118092,6.88753379 Z" id="Shape" fill="#ADB5BD" fill-rule="nonzero"></path>
    </g>
</svg>