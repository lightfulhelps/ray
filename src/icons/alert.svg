

<?xml version="1.0" encoding="UTF-8"?>
<svg width="24px" height="24px" viewBox="0 0 24 24" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <title>alert</title>
    <g id="alert" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
        <g id="icon/green/error-exclamation-mark" transform="translate(-5.000000, -5.000000)" >
            <path d="M12.5,5 C16.6421356,5 20,8.35786438 20,12.5 C20,16.6421356 16.6421356,20 12.5,20 C8.35786438,20 5,16.6421356 5,12.5 C5,8.35786438 8.35786438,5 12.5,5 Z M13.28,15.212 L12.08,15.212 C11.804,15.404 11.636,15.716 11.636,16.052 C11.636,16.628 12.104,17.096 12.668,17.096 C13.256,17.096 13.712,16.628 13.712,16.052 C13.712,15.716 13.544,15.404 13.28,15.212 Z M13.352,8.48 L12.02,8.48 L12.02,11.9 L12.272,14.204 L13.1,14.204 L13.352,11.9 L13.352,8.48 Z" id="Combined-Shape" fill="#ADB5BD" fill-rule="nonzero"></path>
        </g>
    </g>
</svg>