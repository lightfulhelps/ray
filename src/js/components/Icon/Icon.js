// @flow
import React, { Component } from 'react';
import classNames from 'classnames';
import icons from './icons';

type Props = {
  className?: string,
  color?: string,
  hoverColor?: string,
  name: string,
  size?: number,
  style?: { [key: string]: any },
  viewBox?: string,
};

type State = {
  hover: boolean,
};

class Icon extends Component<Props, State> {
  state = {
    hover: false,
  };

  handleMouseEnter = () => {
    this.setState({ hover: true });
  };

  handleMouseLeave = () => {
    this.setState({ hover: false });
  };

  render() {
    const {
      className,
      color = '#adb5bd',
      hoverColor = '#212529',
      name,
      size = 24,
      style = {},
      viewBox = '0 0 24 24',
    } = this.props;
    const classes = classNames(className, 'icon');

    return (
      <svg
        className={classes}
        width={size}
        height={size}
        viewBox={viewBox}
        style={Object.assign({ fill: this.state.hover ? hoverColor : color }, style)}
        onMouseEnter={this.handleMouseEnter}
        onMouseLeave={this.handleMouseLeave}
      >
        {icons[name]}
      </svg>
    );
  }
}

export default Icon;
