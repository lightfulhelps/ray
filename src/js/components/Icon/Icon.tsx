/* eslint-disable no-param-reassign */
import React, { useState } from 'react';
import { nanoid } from 'nanoid';
import classNames from 'classnames';
import icons, { IconNameType } from './icons';

type Props = {
  className?: string;
  color?: string;
  hoverColor?: string;
  isActive?: boolean;
  isDisabled?: boolean;
  name: IconNameType;
  onClick?: (event: React.MouseEvent) => void;
  size?: number;
  style?: {
    [key: string]: string | number;
  };
  theme?: string;
  title?: string;
  viewBox?: string;
  withHover?: boolean;
  [key: string]: any;
};

const Icon: React.FC<Props> = ({
  className,
  color = 'currentColor',
  hoverColor,
  isActive,
  isDisabled,
  name,
  size = '1em',
  style = {},
  theme,
  title,
  viewBox = '0 0 24 24',
  withHover = false,
  ...other
}) => {
  const [hover, setHover] = useState(false);

  const handleMouseEnter = () => {
    setHover(true);
  };

  const handleMouseLeave = () => {
    setHover(false);
  };

  const classes = classNames(
    className,
    'icon',
    theme ? `icon-${theme}` : '',
    withHover ? `with-hover` : '',
    { disabled: isDisabled },
    { active: isActive },
    { 'cursor-pointer': typeof other.onClick === 'function' && !isDisabled }
  );

  const instagramGradientId = theme === 'instagram' ? nanoid() : null;

  if (theme) {
    if (theme === 'instagram') style.fill = `url(#${instagramGradientId})`;
    else delete style.fill;
  } else {
    style.fill = hover && hoverColor ? hoverColor : color;
  }

  return (
    <svg
      {...other}
      className={classes}
      width={size}
      height={size}
      viewBox={viewBox}
      style={style}
      onMouseEnter={handleMouseEnter}
      onMouseLeave={handleMouseLeave}
    >
      {theme === 'instagram' && (
        <defs>
          <radialGradient id={instagramGradientId} r="150%" cx="30%" cy="107%">
            <stop stopColor="#fdf497" offset="0" />
            <stop stopColor="#fdf497" offset="0.05" />
            <stop stopColor="#fd5949" offset="0.45" />
            <stop stopColor="#d6249f" offset="0.6" />
            <stop stopColor="#285AEB" offset="0.9" />
          </radialGradient>
        </defs>
      )}
      {title && <title>{title}</title>}
      {icons[name]}
    </svg>
  );
};

export default Icon;
