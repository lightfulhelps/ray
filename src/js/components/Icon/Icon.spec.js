import React from 'react';
import { shallow } from 'enzyme';
import merge from 'lodash/merge';
import Icon from './Icon';
import icons from './icons';

jest.mock('./icons', () => ({
  analytics: (
    <path
      fillRule="nonzero"
      d="M6.5883 20h-3.727C2.3855 20 2 19.6427 2 19.2017V5.7983C2 5.3573 2.3855 5 2.8613 5h3.727c.4758 0 .8613.3573.8613.7983v13.4034c-.0007.441-.3862.7983-.8613.7983zm7.2749 0h-3.727c-.4759 0-.8613-.3573-.8613-.7983V8.6765c0-.441.3854-.7983.8612-.7983h3.727c.4759 0 .8614.3573.8614.7983v10.5252c0 .441-.3855.7983-.8613.7983zm7.2755 0h-3.727c-.4758 0-.8613-.3573-.8613-.7983V11.258c0-.441.3855-.7983.8613-.7983h3.727c.4758 0 .8613.3573.8613.7983v7.9436c0 .441-.3855.7983-.8613.7983z"
    />
  ),
  comment: (
    <path
      fillRule="nonzero"
      d="M15.2995 5.016c-.0656-.0064-.1312-.0128-.2-.016-.95.0032-1.9.0043-2.85.0032-.2406 0-1.5687 0-3.0812-.0032-.3907.0192-.675.0416-.775.0608-2.872.5149-4.8125 3.406-4.3157 6.3514.4657 2.7664 2.6407 4.6565 5.3844 4.6725.7406.0032 1.4813-.0096 2.2219.0096.1734.006.341.065.4812.1695.7625.6076 1.5063 1.244 2.2688 1.8517.2625.2079.5531.3965.8562.518.5657.224 1.047-.0287 1.2438-.6236a2.651 2.651 0 0 0 .1281-.678 15.506 15.506 0 0 0 .0094-1.3208c-.0094-.2366.0594-.3646.275-.4765 2.1094-1.1002 3.2156-3.1374 3.0344-5.552-.1938-2.552-2.1994-4.73-4.6813-4.9666z"
    />
  ),
}));

const setup = (overrides = {}) => {
  const props = merge(
    {
      name: 'analytics',
      size: 30,
    },
    overrides
  );
  const wrapper = shallow(<Icon {...props} />);

  return {
    wrapper,
    props,
  };
};

describe('<Icon />', () => {
  it('should render', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it('should handle the className prop', () => {
    const { wrapper } = setup({ className: 'custom' });

    expect(wrapper.hasClass('icon')).toBe(true);
    expect(wrapper.hasClass('custom')).toBe(true);
  });

  it('should handle the name prop', () => {
    const { wrapper } = setup({ name: 'analytics' });

    expect(wrapper.contains(icons.analytics)).toBe(true);

    wrapper.setProps({ name: 'comment' });

    expect(wrapper.contains(icons.comment)).toBe(true);
  });

  it('should handle the size prop', () => {
    const { wrapper } = setup({ size: 20 });

    expect(wrapper.prop('width')).toEqual(20);
    expect(wrapper.prop('height')).toEqual(20);
  });

  it('should handle the viewBox prop', () => {
    const { wrapper } = setup({ viewBox: '0 0 16 16' });

    expect(wrapper.prop('viewBox')).toEqual('0 0 16 16');
  });

  it('should update state on mouse enter and mouse leave', () => {
    const { wrapper } = setup();

    expect(wrapper.state('hover')).toBe(false);

    wrapper.simulate('mouseEnter');

    expect(wrapper.state('hover')).toBe(true);

    wrapper.simulate('mouseLeave');

    expect(wrapper.state('hover')).toBe(false);
  });

  it('should handle the style prop', () => {
    const { wrapper } = setup({ style: { marginTop: '5px' } });

    expect(wrapper.prop('style')).toEqual({
      fill: 'currentColor',
      marginTop: '5px',
    });
  });

  it('should handle the color prop', () => {
    const { wrapper } = setup({ color: '#ffff00' });

    expect(wrapper.prop('style')).toEqual({
      fill: '#ffff00',
    });
  });

  it('should handle the theme prop', () => {
    const { wrapper } = setup();

    expect(wrapper.hasClass('icon-primary')).toBe(false);

    wrapper.setProps({ theme: 'primary' });

    expect(wrapper.hasClass('icon-primary')).toBe(true);
  });

  it('should not use color if theme is provided', () => {
    const { wrapper } = setup({ color: 'blue' });

    expect(wrapper.prop('style').fill).toBe('blue');

    wrapper.setProps({ theme: 'primary' });

    expect(wrapper.prop('style').fill).toBe(undefined);
  });

  it('should not use hoverColor if theme is provided', () => {
    const { wrapper } = setup({ hoverColor: 'blue' });

    wrapper.setState({ hover: true });

    expect(wrapper.prop('style').fill).toBe('blue');

    wrapper.setProps({ theme: 'primary' });

    expect(wrapper.prop('style').fill).toBe(undefined);
  });

  it('should not update fill color on hover if hoverColor is not provided', () => {
    const { wrapper } = setup({ color: 'red', hoverColor: null });

    expect(wrapper.prop('style')).toEqual({ fill: 'red' });

    wrapper.simulate('mouseEnter');

    expect(wrapper.prop('style')).toEqual({ fill: 'red' });

    wrapper.simulate('mouseLeave');

    expect(wrapper.prop('style')).toEqual({ fill: 'red' });
  });

  it('should update fill color on hover if hoverColor is provided', () => {
    const { wrapper } = setup({ color: 'red', hoverColor: 'blue' });

    expect(wrapper.prop('style')).toEqual({ fill: 'red' });

    wrapper.simulate('mouseEnter');

    expect(wrapper.prop('style')).toEqual({ fill: 'blue' });

    wrapper.simulate('mouseLeave');

    expect(wrapper.prop('style')).toEqual({ fill: 'red' });
  });

  it('should pass through other props', () => {
    const { wrapper } = setup({ tabIndex: 1, id: 'test' });

    expect(wrapper.prop('tabIndex')).toEqual(1);
    expect(wrapper.prop('id')).toEqual('test');
  });
});
