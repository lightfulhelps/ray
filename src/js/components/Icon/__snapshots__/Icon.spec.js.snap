// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> should render 1`] = `
<svg
  className="icon"
  height={30}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "fill": "#ff0000",
    }
  }
  viewBox="0 0 24 24"
  width={30}
>
  <g
    fillRule="nonzero"
  >
    <path
      d="M-2-5h24v24H-2z"
      fill="none"
    />
    <path
      d="M4.5883 15H.8613C.3855 15 0 14.6427 0 14.2017V.7983C0 .3573.3855 0 .8613 0h3.727c.4758 0 .8613.3573.8613.7983v13.4034c-.0007.441-.3862.7983-.8613.7983zm7.2749 0h-3.727c-.4759 0-.8613-.3573-.8613-.7983V3.6765c0-.441.3854-.7983.8612-.7983h3.727c.4759 0 .8614.3573.8614.7983v10.5252c0 .441-.3855.7983-.8613.7983zm7.2755 0h-3.727c-.4758 0-.8613-.3573-.8613-.7983V6.258c0-.441.3855-.7983.8613-.7983h3.727c.4758 0 .8613.3573.8613.7983v7.9436c0 .441-.3855.7983-.8613.7983z"
    />
  </g>
</svg>
`;
